### ISSUE TITLE
adding type aliases for the block height and unix time instead of just using a `u32` and `u64`? It would make the code easier to read and I don't think there are any drawbacks. Could put them in `chain/src/lib.rs`.

### ISSUE THREAD

**Issue by LLFourn** [bitcoindevkit/bdk#1016]
A little off-topic for this PR, but how'd you'all feel about adding type aliases for the block height and unix time instead of just using a `u32` and `u64`? It would make the code easier to read and I don't think there are any drawbacks. Could put them in `chain/src/lib.rs`.

```rust
// The bitcoin block height, 0 is the genesis block.
pub type BlockHeight = u32;

// Number of seconds that have elapsed since 00:00:00 UTC on 1 January 1970, the Unix epoch.
pub type UnixSeconds = u64;
```

_Originally posted by @notmandatory in https://github.com/bitcoindevkit/bdk/issues/1002#issuecomment-1602955088_

**Comment by ValuedMammal** [comment]
I think once this is done, the satoshi `u64` will be one of the last remaining holdouts that refuses to have a friendly type.

**Comment by ValuedMammal** [comment]
`UnixSeconds` can be a tuple struct that holds a `u64`

**Comment by 15IITian** [comment]
Should we make them tuple struct or not?

**Comment by apoelstra** [comment]
You may want to use (or alias) this type from rust-bitcoin https://docs.rs/bitcoin/latest/bitcoin/blockdata/locktime/absolute/struct.Height.html and/or the corresponding `Time` structure.

For seconds there is also `std::time::Duration` for some situations.

**Comment by 15IITian** [comment]
so should I direcly use them or create an alias with the name given above.
( For `BlockHeight` -> seems  to directly use the struct but what to do for `UnixSecond` -> as this term is often used in the codebase ) ?

**Comment by notmandatory** [comment]
I agree we should use rust-bitcoin `BlockHeight` and that for `UnixSeconds` it might be better to wrap the u64, unless it's already done somewhere in the core lib (for no-std compatibility) but I couldn't find anything.  Good summary of the pattern here: https://effective-rust.com/newtype.html.

**Comment by 15IITian** [comment]
> I agree we should use rust-bitcoin `BlockHeight` 

@notmandatory 
The `Height` struct is wrapped over `u16` in the libray but as per the issue -> we need u32 ?

**Comment by notmandatory** [comment]
@15IITian which `Time` struct do you mean? The closest equivalent to what I think we need in BDK is [`block::Header.time`](https://docs.rs/bitcoin/latest/bitcoin/blockdata/block/struct.Header.html#structfield.time) which is a `u32`. But even that isn't exactly what we're tracking in the BDK chain structures which track first time WE saw the data (not what a miner put in the block header). In that case I think it makes sense to create our own `UnixSeconds(u64)` struct. This also avoids running into any `u32` [year 2038 (+68) issues](https://en.bitcoin.it/wiki/Block_timestamp) (if BDK is still being used in 2106).

**Comment by 15IITian** [comment]
Actually the description of `UnixSecond` given above matches with the `Time`  struct in rust-bitcoin .
Thus looks that both  perform the same functionality but uses different data types -> so therefore  I  asked that are we needed to use `u64`.

**Comment by apoelstra** [comment]
> This also avoids running into any `u32` [year 2038 (+68) issues](https://en.bitcoin.it/wiki/Block_timestamp) (if BDK is still being used in 2106)

As commented in [BIP 65 OP_CSV](https://github.com/bitcoin/bips/blob/master/bip-0065.mediawiki) Bitcoin itself has a year-2106 problem in its `locktime` field. So it may make sense to replicate that problem here rather than allowing users to create transactions that appear to be valid but are actually consensus-invalid.

**Comment by 15IITian** [comment]
> @15IITian which `Time` struct do you mean? The closest equivalent to what I think we need in BDK is [`block::Header.time`](https://docs.rs/bitcoin/latest/bitcoin/blockdata/block/struct.Header.html#structfield.time) which is a `u32`.

@notmandatory 
Sorry I now realise that you were to referring to different `Time` alias type.
Here's  `Time struct` 
(https://docs.rs/bitcoin/latest/bitcoin/blockdata/locktime/absolute/struct.Time.html)
which I was talking about.

and gives the same functionality as we needs.

**Comment by notmandatory** [comment]
For building transactions with a `locktime` I completely agree we should continue to use the `rust-bitcoin` `absolute::LockTime` enum which uses the `absolute::Time(u32)` struct.  

The `UnixSeconds()` struct I have in mind is only used for internal timestamps in the`bdk_chain` crate, eg. the last time we saw a Tx in the mempool. That said, the `rust-bitcoin` `locktime::absolute::Time` structure would also work. But the goals of this issue is to make the code easier to read and I think using a `locktime` module related struct for timestamps unrelated to locktimes is a potentially confusing.

**Comment by 15IITian** [comment]
So means I have to absolute:Time struct outside the `bdk-chain` crate and our custom struct inside it, right?

**Comment by 15IITian** [comment]
Should the `BlockHeight` and `UnixSecond` struct have default value as 0?

**Comment by LLFourn** [comment]
I agree we should distinguish between `u64` which is the timestamp we last saw a transaction vs actual locktime. I think we could use the rust-bitcoin definition for actual chain confirmation time though since that's what's going to need to be converted to do relative lock time checks with miniscript etc.
